lib/Serialization/Deserialization.cpp

```diff
@@ -4532,12 +4543,6 @@ class DeclDeserializer {
 
 #ifndef NDEBUG
     if (outerParams) {
-      unsigned paramCount = 0;
-      for (auto *paramList = outerParams;
-           paramList != nullptr;
-           paramList = paramList->getOuterParameters()) {
-        paramCount += paramList->size();
-      }
       assert(paramCount ==
```

Comment:
This assert should be moved as well, since it's the reason we're computing `paramCount` here at all.